jiant/scripts/download_data/datasets/files_tasks.py

```diff
@@ -222,3 +241,386 @@ def download_qasrl_data_and_write_config(
+                if answer == "none":
+                    answer_item["noAnswer"] = True
+                elif answer == "bad_question":
+                    answer_item["badQuestion"] = True
+                else:
+                    s, e = map(int, answer.split(":"))
+                    answer_item["s"] = s
+                    answer_item["e"] = e
+                q["validatedAnswers"].append(answer_item)
+        consensus_start, consensus_end = get_consensus_answer(row)
+        if consensus_start is None and consensus_end is None:
+            if q.get("isQuestionBad", 0) >= 0.5:
+                q["consensus"] = dict(badQuestion=True)
+            else:
+                q["consensus"] = dict(noAnswer=True)
+        else:
+            q["consensus"] = dict(s=consensus_start, e=consensus_end)
+        questions.append(q)
+
+    phase_dict = {
+        "train": [],
+        "val": [],
+        "test": [],
+    }
+    phase_map = {"train": "train", "dev": "val", "test": "test"}
+    for entry in data:
+        phase = phase_map[entry["type"]]
+        output_entry = {"text": entry["text"], "storyId": entry["storyId"], "qas": []}
+        for qn in entry["questions"]:
+            if "badQuestion" in qn["consensus"] or "noAnswer" in qn["consensus"]:
+                continue
+            output_entry["qas"].append({"question": qn["q"], "answer": qn["consensus"]})
+        phase_dict[phase].append(output_entry)
+    for phase, phase_data in phase_dict.items():
+        py_io.write_jsonl(phase_data, os.path.join(task_data_path, f"{phase}.jsonl"))
+    py_io.write_json(
+        data={
+            "task": task_name,
+            "paths": {
+                "train": os.path.join(task_data_path, "train.jsonl"),
+                "val": os.path.join(task_data_path, "val.jsonl"),
+                "test": os.path.join(task_data_path, "val.jsonl"),
+            },
+            "name": task_name,
+        },
+        path=task_config_path,
+    )
+    for file_name in file_name_list:
+        os.remove(os.path.join(task_data_path, file_name))
+
+
+def download_mrqa_natural_questions_data_and_write_config(
+    task_name: str, task_data_path: str, task_config_path: str
+):
+    os.makedirs(task_data_path, exist_ok=True)
+    download_utils.download_file(
+        "https://s3.us-east-2.amazonaws.com/mrqa/release/v2/train/NaturalQuestionsShort.jsonl.gz",
+        os.path.join(task_data_path, "train.jsonl.gz"),
+    )
+    download_utils.download_file(
+        "https://s3.us-east-2.amazonaws.com/mrqa/release/v2/dev/NaturalQuestionsShort.jsonl.gz",
+        os.path.join(task_data_path, "val.jsonl.gz"),
+    )
+    py_io.write_json(
+        data={
+            "task": task_name,
+            "paths": {
+                "train": os.path.join(task_data_path, "train.jsonl.gz"),
+                "val": os.path.join(task_data_path, "val.jsonl.gz"),
+            },
+            "name": task_name,
+        },
+        path=task_config_path,
```

Comment:
I think it would be better to extract the file before generating the config (this adds another potential file format for tasks to consume)

Reply:
The unzipped version is 3x larger. Also QA-SRL currently also uses `gzip` during the data loading process. Since reading `.jsonl.gz` data is in the same format as `.jsonl`, I think it's not too problematic to keep it as is.